include/unifex/task.hpp

```diff
@@ -314,8 +331,9 @@ struct _promise final {
     }
 
     template <typename Value>
+    // todo: consider if this should be nothrow or not
     decltype(auto) await_transform(Value&& value) {
-      if constexpr (is_sender_for_v<remove_cvref_t<Value>, schedule>) {
+      if constexpr (!nothrow && is_sender_for_v<remove_cvref_t<Value>, schedule>) {
```

Comment:
I think we should find a way to warn folks that no-throw tasks don't support a magic `co_await schedule(…)`.

Maybe, instead of adding `!nothrow` to the condition here, add a `static_assert(!nothrow, "Magic rescheduling isn't supported by no-throw tasks")`, or something, inside the `if constexpr` block.

What do you think?

Reply:
Won't this trigger an std::terminate if a nothrow task calls schedule()? I think schedule() in a nothrow task should still be valid behavior, even if it ends up behaving differently than schedule() in a regular task. 

i can add a comment block here calling this out for nothrow task, and add it explicitly to our documentation.

Reply:
> Won't this trigger an std::terminate if a nothrow task calls schedule()?

No, it'd fail to compile.

> I think schedule() in a nothrow task should still be valid behavior, even if it ends up behaving differently than schedule() in a regular task. What do you think?

I'm torn. Long term, I agree it should be valid, but I'm worried about people switching from `task<T>` to `nothrow_task<T>` (especially us, internally) and accidentally changing the scheduling behaviour of their code.

If you want to change this coroutine from `task<int>` to `nothrow_task<int>` and you want to _keep_ the reschedule for some reason, you can just pass it through a `then` to hide its type:
```
task<int> expensiveInt() {
  co_await schedule(s);

  co_return 42;
}
```
can be changed to
```
nothrow_task<int> expensiveInt() {
  // still reschedules but doesn't change the current scheduler
  co_await then(schedule(s), []() noexcept {});

  co_return 42;
}
```

In the farther future, I expect to unwind the `sender_for` machinery that Eric added to be able to detect `co_await schedule(s);`. To do that, we'll have to invent a new way to change the current coroutine's current scheduler (something like `co_await reschedule_current_coroutine(s);`), which we can ensure supports both `task<>` and `nothrow_task<>`. Once we've done that, we can delete the magic in `task<>` and remove the `static_assert` in `nothrow_task<>`.